packages/sourcecred/src/plugins/external/plugin.js

```diff
@@ -0,0 +1,203 @@
+// @flow
+
+import type {Plugin, PluginDirectoryContext} from "../../api/plugin";
+import {
+  type PluginDeclaration,
+  type PluginDeclarationJSON,
+  fromJSON as declarationFromJSON,
+} from "../../analysis/pluginDeclaration";
+import {join as pathJoin} from "path";
+import {type TaskReporter} from "../../util/taskReporter";
+import type {ReferenceDetector} from "../../core/references";
+import {
+  type WeightedGraph,
+  fromJSON as weightedGraphFromJSON,
+  type WeightedGraphJSON,
+} from "../../core/weightedGraph";
+import {loadJson, loadJsonWithDefault} from "../../util/storage";
+import {merge as mergeWeights} from "../../core/weights";
+import {weightsForDeclaration} from "../../analysis/pluginDeclaration";
+import {type PluginId} from "../../api/pluginId";
+import {ZipStorage} from "../../core/storage/zip";
+import {
+  type IdentityProposal,
+  identityProposalsParser,
+} from "../../core/ledger/identityProposal";
+import * as Combo from "../../util/combo";
+import {type DataStorage} from "../../core/storage";
+import {NetworkStorage} from "../../core/storage/networkStorage";
+import {declaration as defaultDeclaration} from "./defaultDeclaration";
+import * as C from "../../util/combo";
+
+export const ExternalPluginIdOwner = "external";
+
+const FileNames = {
+  GRAPH: "graph.json",
+  GRAPH_ZIP: "graph.json.gzip",
+  DECLARATION: "declaration.json",
+  IDENTITIES: "identityProposals.json",
+};
+type FileType = $Keys<typeof FileNames>;
+
+type ExternalPluginConfig = {|
+  +graphUrl: string,
+  +declarationUrl?: string,
+  +identityProposalsUrl?: string,
+|};
```

Comment:
I'd like to see a URL parser created for these, but that can probably be a separate issue, or even a good first issue. I thought we had one, but I can't find it. It'd be really good to "fail fast" on something like this before any network requests are made.

Reply:
so true. create an issue?